microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -138,44 +139,33 @@ func reconcileResource(step string, ns string, svc string, svcCfg *map[string]st
 		return fmt.Errorf("failed to apply user config: %v", err)
 	}
 	resources = strings.Split(appliedCfg, "---")
-	fmt.Printf("The raw yaml file has been splitted into %v yaml files", len(resources))
-	decUnstructured := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	fmt.Printf("The raw yaml file has been split into %v yaml files\n", len(resources))
 
 	for _, res := range resources {
 		if res == "" || !strings.Contains(res, "kind:") {
 			continue
 		}
 
-		obj := &unstructured.Unstructured{}
-		_, gvk, err := decUnstructured.Decode([]byte(res), nil, obj)
-		if err != nil {
-			return fmt.Errorf("failed to decode YAML: %v", err)
-		}
-
-		gvr, _ := meta.UnsafeGuessKindToResource(*gvk)
+		// if create failed, wait 2s and retry, this will be removed when the monitor task is implemented
 		for {
-			patchBytes, err := json.Marshal(obj)
-			if err != nil {
-				return fmt.Errorf("failed to marshal config to json: %v", err)
-			}
+			createdObj, err := applyResourceToK8s(ctx, dynamicClient, ns, []byte(res))
 
-			createdObj, err := dynamicClient.Resource(gvr).Namespace(ns).Patch(context.TODO(), obj.GetName(), types.ApplyPatchType, patchBytes, metav1.PatchOptions{
-				FieldManager: "gmc-controller",
-				Force:        ptr.To(true),
-			})
 			if err != nil {
 				fmt.Printf("Failed to reconcile resource: %v\n", err)
 			} else {
-				fmt.Printf("Resource %s/%s created\n", gvk.Kind, createdObj.GetName())
+				fmt.Printf("Success to reconcile %s: %s\n", createdObj.GetKind(), createdObj.GetName())
+
+				// return the service obj to get the serivce URL from it
 				if retSvc != nil && createdObj.GetKind() == "Service" {
 					err = scheme.Scheme.Convert(createdObj, retSvc, nil)
 					if err != nil {
 						fmt.Printf("Failed to save service: %v\n", err)
 					}
 				}
+
 				break
 			}
-			time.Sleep(time.Second * 2)
+			time.Sleep(2 * time.Second)
```

Comment:
Any benefit to having the 2 first in the sleep argument?

Reply:
no benefit, just remove the line and add it back when implementing the PR.